mobile/src/components/common/search/search.tsx

```diff
@@ -0,0 +1,52 @@
+import { debounce } from 'debounce';
+import React, { FC } from 'react';
+import { TextInput } from 'react-native';
+
+import { AppColor } from '~/common/enums/enums';
+import { Icon, View } from '~/components/common/common';
+import { useEffect, useState } from '~/hooks/hooks';
+
+import { styles } from './styles';
+
+type Props = {
+  onSearch: (text: string) => void;
+};
+
+const Search: FC<Props> = ({ onSearch }) => {
+  const [text, setText] = useState('');
+  const [borderColor, setBorderColor] = useState('transparent');
+  const handleOnSearch = debounce(onSearch, 1000);
```

Comment:
As for debounce - please create a helper in shared folder, we will be use in frontend and mobile as well